.dotnet/src/Custom/Files/IncrementalUploadJob.cs

```diff
@@ -0,0 +1,101 @@
+
+    /// <summary>
+    /// A token that can be serialized/deserialized and used to reinstantiate an <see cref="IncrementalUploadJob"/>,
+    /// including to facilitate parallel data part uploads across multiple environments.
+    /// </summary>
+    /// <remarks>
+    /// The full binary representation of the original server <c>Upload</c> is also represented in this token via
+    /// <see cref="ContinuationToken.ToBytes"/>.
+    /// </remarks>
+    public ContinuationToken RehydrationToken { get; }
+
+    /// <summary>
+    /// Creates a new instance of <see cref="IncrementalUploadJob"/> based on a live service response.
+    /// </summary>
+    /// <param name="internalJobInfo"></param>
+    /// <param name="internalClient"></param>
+    /// <param name="protocolJobResponse"></param>
+    internal IncrementalUploadJob(InternalUploadJobInfo internalJobInfo, InternalUploadClient internalClient, PipelineResponse protocolJobResponse)
+        : base(protocolJobResponse)
+    {
+        _internalJobInfo = internalJobInfo;
+        _internalClient = internalClient;
+        RehydrationToken = new InternalIncrementalUploadJobToken(internalJobInfo);
+    }
+
+    /// <summary>
+    /// Creates a new instance of <see cref="IncrementalUploadJob"/> from a rehydration <see cref="ContinuationToken"/>.
+    /// </summary>
+    /// <param name="rehydrationToken"></param>
+    /// <param name="internalClient"></param>
+    internal IncrementalUploadJob(ContinuationToken rehydrationToken, InternalUploadClient internalClient)
+        : base()
+    {
+        InternalIncrementalUploadJobToken specificToken = InternalIncrementalUploadJobToken.FromToken(rehydrationToken);
+
+        _internalJobInfo = specificToken.JobInfo;
+        _internalClient = internalClient;
+        RehydrationToken = specificToken;
+    }
+
+    /// <summary>
+    /// Instantiates a new instance of <see cref="IncrementalUploadJob"/> that will interact with the same server
+    /// <c>Upload</c> object created by a previous call to
+    /// <see cref="FileClient.CreateIncrementalUploadJob(BinaryContent, RequestOptions)"/>. A rehydrated instance may
+    /// be used to upload data parts and complete the job in a different environment relative to where the job was
+    /// started.
+    /// </summary>
+    /// <param name="client"> The <see cref="FileClient"/> instance that the rehydrated job should use for operations. </param>
+    /// <param name="rehydrationToken">
+    /// The token, retrieved from the <see cref="IncrementalUploadJob.RehydrationToken"/> property, that will configure
+    /// the rehydrated job instance.
+    /// </param>
+    /// <returns>
+    /// A new instance of <see cref="IncrementalUploadJob"/> that will interact with the same <c>Upload</c> that the
+    /// previous job, represented by <paramref name="rehydrationToken"/>, does.
+    /// </returns>
+    public static IncrementalUploadJob Rehydrate(FileClient client, ContinuationToken rehydrationToken)
```

Comment:
Rehydrate is terminology we have in Azure.Core for LROs -- this one is not an LRO, so it would be reasonable to use continue verbiage here if desired - I defer to others, and may have a slight preference for not using "rehydrate" to potentially reduce confusion around this being an LRO.  Krzysztof reminded me yesterday that this looks a lot like the Block Blobs API in Azure storage - there may be naming conventions there we could follow.

Reply:
I prefer `FromContinuationToken`, too, so I'm going to make that change. We can review it as a standardization pass later on, of course.